src/main/java/com/wanderersoftherift/wotr/core/rift/RiftLevelManager.java

```diff
@@ -35,33 +36,86 @@
 import net.neoforged.neoforge.network.PacketDistributor;
 import net.neoforged.neoforge.server.ServerLifecycleHooks;
 import org.apache.commons.io.FileUtils;
+import org.jetbrains.annotations.Nullable;
 
-import javax.annotation.Nullable;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.util.List;
 import java.util.Optional;
 import java.util.Random;
+import java.util.Set;
 
-public class RiftLevelManager {
+/**
+ * Static manager for handing access to, creation, and destruction of a rift
+ */
+public final class RiftLevelManager {
+
+    private RiftLevelManager() {
+    }
 
     /**
      * @param id
      * @return Whether a level with the given id exists
      */
     public static boolean levelExists(ResourceLocation id) {
         var server = ServerLifecycleHooks.getCurrentServer();
-        var existingRift = server.forgeGetWorldMap().get(ResourceKey.create(Registries.DIMENSION, id));
-        return existingRift != null;
+        var dimension = server.forgeGetWorldMap().get(ResourceKey.create(Registries.DIMENSION, id));
+        return dimension != null;
+    }
+
+    /**
+     * @param id
+     * @return The rift level with the given id, if it exists
+     */
+    public static @Nullable ServerLevel getRiftLevel(ResourceLocation id) {
+        var server = ServerLifecycleHooks.getCurrentServer();
+
+        ServerLevel serverLevel = server.forgeGetWorldMap().get(ResourceKey.create(Registries.DIMENSION, id));
+        if (serverLevel != null && RiftData.isRift(serverLevel)) {
+            return serverLevel;
+        }
+        return null;
+    }
+
+    /**
+     * @param player The player to remove from a rift level
+     * @return Whether the player was successfully removed from a rift
+     */
+    public static boolean removePlayerFromRift(ServerPlayer player) {
```

Comment:
Should we use this method also in the tick method in RiftEvents?

Reply:
No, because this method is for players leaving the rifts alive - the behaviour doesn't mesh well (break inventory snapshots for one). Naming could be better. I think I can make things cleaner though.

Reply:
Renamed to `returnPlayerFromRift`. It is alongside an onPlayerDeath method.